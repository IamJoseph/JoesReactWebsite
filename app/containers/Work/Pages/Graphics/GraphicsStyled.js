import styled from 'styled-components';

export const ImgContainer = styled.div`
  max-width: 655px;
  &.clicked{
    background-color: #fff;
    bottom: 0;
    height: 100%;
    left: 0;
    margin: 0;
    max-width: 100%;
    position: fixed;
    right: 0;
    text-align: center;
    top: 0;
    width: 100%;
    img{
      margin-top: 80px;
      max-height: calc(100% - 80px);
      max-width: 100%;
    }
  }
`;

export const IMG = styled.img`
  cursor: pointer;
  margin-top: 2.5em;
  max-width: 100%;
`;
