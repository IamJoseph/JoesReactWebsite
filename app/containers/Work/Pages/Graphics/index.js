import React, { PureComponent } from 'react';
import { IMG, ImgContainer } from './GraphicsStyled';
import { Container, UL, H2, LI } from '../PagesStyled';
import bricksTee from '../../../../assets/bricks.png';
import cometTee from '../../../../assets/comet.png';
import museumTee from '../../../../assets/museum.png';
import officeDogTee from '../../../../assets/office_dog.png';
import toggle from '../toggleFullScreen';

export default class Graphics extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      bricks: false,
      comet: false,
      museum: false,
      officeDog: false,
    };
  }

  toggleFullScreen = toggle;

  render() {
    const { bricks, comet, museum, officeDog } = this.state;
    return (
      <Container>
        <UL>
          <H2>
            Graphics
          </H2>
          <LI>
            Conceptual vectored tee designs
          </LI>
          <ImgContainer onClick={() => this.toggleFullScreen('bricks')} className={bricks && 'clicked'}>
            <IMG src={bricksTee} alt="brick red style graphic tee" />
          </ImgContainer>
          <ImgContainer onClick={() => this.toggleFullScreen('comet')} className={comet && 'clicked'}>
            <IMG src={cometTee} alt="floating island abstract light grey graphic tee" />
          </ImgContainer>
          <ImgContainer onClick={() => this.toggleFullScreen('museum')} className={museum && 'clicked'}>
            <IMG src={museumTee} alt="museum dark grey graphic tee" />
          </ImgContainer>
          <ImgContainer onClick={() => this.toggleFullScreen('officeDog')} className={officeDog && 'clicked'}>
            <IMG src={officeDogTee} alt="blue office worker dog graphic tee" />
          </ImgContainer>
        </UL>
      </Container>
    );
  }
}
