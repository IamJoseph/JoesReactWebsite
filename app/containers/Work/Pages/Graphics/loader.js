/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Graphics')
    .then(cb)
    .catch(errorLoading);
};
