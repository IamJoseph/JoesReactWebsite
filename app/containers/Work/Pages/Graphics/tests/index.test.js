import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Graphics from '../index';
import { ImgContainer } from '../GraphicsStyled';

describe('<Graphics />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Graphics />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on 1st image container when clicked', () => {
    const wrapper = shallow(<Graphics />);
    wrapper.find(ImgContainer).at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on 2nd image container when clicked', () => {
    const wrapper = shallow(<Graphics />);
    wrapper.find(ImgContainer).at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on 3rd image container when clicked', () => {
    const wrapper = shallow(<Graphics />);
    wrapper.find(ImgContainer).at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on 4th image container when clicked', () => {
    const wrapper = shallow(<Graphics />);
    wrapper.find(ImgContainer).at(3).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
