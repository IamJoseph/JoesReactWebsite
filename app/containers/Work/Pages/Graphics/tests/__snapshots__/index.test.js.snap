// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Graphics /> should render correctly 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Graphics
    </styled.h2>
    <styled.li>
      Conceptual vectored tee designs
    </styled.li>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="brick red style graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="floating island abstract light grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="museum dark grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="blue office worker dog graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Graphics /> should toggle className clicked on 1st image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Graphics
    </styled.h2>
    <styled.li>
      Conceptual vectored tee designs
    </styled.li>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="brick red style graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="floating island abstract light grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="museum dark grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="blue office worker dog graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Graphics /> should toggle className clicked on 2nd image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Graphics
    </styled.h2>
    <styled.li>
      Conceptual vectored tee designs
    </styled.li>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="brick red style graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="floating island abstract light grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="museum dark grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="blue office worker dog graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Graphics /> should toggle className clicked on 3rd image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Graphics
    </styled.h2>
    <styled.li>
      Conceptual vectored tee designs
    </styled.li>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="brick red style graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="floating island abstract light grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="museum dark grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="blue office worker dog graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Graphics /> should toggle className clicked on 4th image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Graphics
    </styled.h2>
    <styled.li>
      Conceptual vectored tee designs
    </styled.li>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="brick red style graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="floating island abstract light grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="museum dark grey graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="blue office worker dog graphic tee"
        src="IMAGE_MOCK"
      />
    </styled.div>
  </styled.ul>
</styled.section>
`;
